src/wallet/walletdb.cpp

```diff
@@ -726,7 +732,7 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
             {
                 // losing keys is considered a catastrophic error, anything else
                 // we assume the user can live with:
-                if (IsKeyType(strType) || strType == DBKeys::DEFAULTKEY) {
+                if (wss.corrupt || IsKeyType(strType) || strType == DBKeys::DEFAULTKEY) {
```

Comment:
A corrupt/missing transaction isn't supposed to follow this path, but rather the one below which simply triggers a rescan.